src/RoatpProviderModeration/SFA.DAS.RoatpProviderModeration.Application/Queries/GetProvider/GetProviderQueryHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.RoatpProviderModeration.Application.InnerApi.Requests;
+using SFA.DAS.RoatpProviderModeration.Application.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Net;
+
+namespace SFA.DAS.RoatpProviderModeration.Application.Queries.GetProvider
+{
+    public class GetProviderQueryHandler : IRequestHandler<GetProviderQuery, GetProviderResult>
+    {
+        private readonly IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _courseManagementApiClient;
+        private readonly ILogger<GetProviderQueryHandler> _logger;
+
+        public GetProviderQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> courseManagementApiClient, ILogger<GetProviderQueryHandler> logger)
+        {
+            _courseManagementApiClient = courseManagementApiClient;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderResult> Handle(GetProviderQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider request received for ukprn {ukprn}", request.Ukprn);
+
+            var response = await _courseManagementApiClient.GetWithResponseCode<GetProviderResponse>(new GetProviderRequest(request.Ukprn));
+            if (response.StatusCode != HttpStatusCode.OK && response.StatusCode != HttpStatusCode.NotFound)
```

Comment:
NotFound should return null isn't it?

Reply:
I see what is being done here. Cleaner way of doing things would be positive checks so it is obvious. 
```
if (response.StatusCode == HttpStatusCode.Ok) return response.Body;
if (response.StatusCode == HttpStatusCode.NotFound) return null; 
/// else unexpected response throw exception
```